subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyModifier.java

```diff
@@ -88,7 +96,8 @@ default <D extends ModuleDependency> Provider<D> modify(Provider<D> providerToDe
      * @param dependency the dependency to modify
      * @return the modified dependency
      * @param <D> the type of the {@link ModuleDependency}
-     * Implement this method.
+     * @implSpec This method must be implemented.
+     * @implNote All other implementations of {@code modify(...)} delegate to this method.
```

Comment:
🤔 I think this is ought to be in the main javadoc paragraph or `@implSpec`.  It's not a note about something interesting about the implementation (there is no implementation), it's about something to consider during implementation.

Reply:
Do you mean this would make a better class-level note?

Reply:
Maybe.  I was mostly thinking that this was not appropriate use for `@implNote` (as this is an abstract method, so there is no implementation to note about).

Reply:
PTAL at the class-level version of this now.

I was thinking the implNote is used for "interesting information that's not necessary for implementing the method" and implSpec is used for "things you need to know when you implement this method".

Reply:
I had the idea that it was more `@implSpec` "here's things you need to do when you implement X" vs. `@implNote` "here's interesting things about how this implements X"

Reply:
I think we're saying almost the same thing:
`@implSpec` "here's things you need to do when you implement X" == "things you need to know when you implement this method".
`@implNote` "here's interesting things about how this implements X" == "interesting information that's not necessary for implementing the method"

I agree that implNote on a method without an implementation is weird.